### PR TITLE
Modernize instructor course admin instances

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to convert the `instructorCourseAdminInstances` page from JavaScript and EJS to TypeScript and HTML template literals.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ router.get(
       const row = enrollmentCounts.find((row) => idsEqual(row.course_instance_id, ci.id));
       ci.number = row?.number || 0;
     });
-    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    res.send(InstructorCourseAdminInstances({ resLocals: res.locals }));
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts:46]: There are some things we're attaching to `res.locals` above that could instead be passd explicitly to `InstructorCourseAdminInstances` via separate properties.

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            <div class="table-responsive">
+              <table class="table table-sm table-hover table-striped">
+                <thead>
+                  <tr>
+                    <th>Long Name</th>
+                    <th>CIID</th>
+                    <th id="earliest-access-date">
+                      Earliest Access Date
+                      <a
+                        role="button"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts:73]: Can we replace `<a role="button" tabindex="0" ...>` with an actual `<button>`?

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts:1]: I can't comment on the lines directly, but there are a few non-Zod-validated queries here; it should be pretty straightforward to add Zod validation to them.

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts]: Moved `courseInstances` from `res.locals` to a local variable in [86e4395](https://github.com/PrairieLearn/PrairieLearn/pull/9812/commits/86e4395ef9e0858a540dadab76ffab2835b365d8).

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts (hunk 2 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts]: Replaced `<a role="button" tabindex="0" ...>` with a `<button>` here and below in [86e4395](https://github.com/PrairieLearn/PrairieLearn/pull/9812/commits/86e4395ef9e0858a540dadab76ffab2835b365d8).

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts:1]: Added Zod validation to the `select_course_instance_id_from_uuid` query in [86e4395](https://github.com/PrairieLearn/PrairieLearn/pull/9812/commits/86e4395ef9e0858a540dadab76ffab2835b365d8). `selectCourseInstancesWithStaffAccess` is using a Zod-validated query in the `course-instances` model. Are there remaining non-Zod-validated queries that I am missing?

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts [hunk 3]
```diff
@@ -37,13 +40,13 @@ router.get(
     const enrollmentCounts = await sqldb.queryRows(
       sql.select_enrollment_counts,
       { course_id: res.locals.course.id },
-      z.object({ course_instance_id: CourseInstanceSchema.shape.id, number: z.string() }),
+      z.object({ course_instance_id: CourseInstanceSchema.shape.id, number: z.number() }),
     );
-    res.locals.course_instances.forEach((ci) => {
+    courseInstances.forEach((ci) => {
       const row = enrollmentCounts.find((row) => idsEqual(row.course_instance_id, ci.id));
       ci.number = row?.number || 0;
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts:47]: "number" is extremely vague. As long as we're changing things here, can we call this `enrollment_count` instead?

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts:1]: Nope, your changes look good to me!

File: apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts (hunk 3 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.ts]: Updated to `enrollment_count` in [4708ba6](https://github.com/PrairieLearn/PrairieLearn/pull/9812/commits/4708ba69e4b442865ec896dec7a5155fab7d73b4).

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**